spec/defines/openldap_server_globalconf_spec.rb

```diff
@@ -9,25 +9,18 @@
         facts
       end
 
+      let :pre_condition do
+        "class { 'openldap::server': }"
```

Comment:
This should not be needed. The defined type should explicitly include the classes it depends on.

Reply:
Yes sure, but for now, this is something that would improve something that would improve something that I am tweaking to fix a bug :unamused:.

So much sadness in a single module :sob:

Reply:
Hum, some (many) classes have this in the preamble:

```puppet
   if ! defined(Class['openldap::server']) {
     fail 'class openldap::server has not been evaluated'
   }
```

Not sure if I should add this for consistency rather than auto-including the class…  Though?

Reply:
This is now out-of-scope of this PR, but I gues we can still improve this point in another PR :wink:

Reply:
That code above makes me 😭 No need for ifs and fails. If it is meant to be there just include it.

It is common that authors assume if you use a defined type that you also included the main class and set themselves up for an implicit dependency which surfaces when you add a simple spec test for the defined type and it cannot compile because the dependencies to the class are missing. Simple fix is just `include class`.

Reply:
Let's handle this in #322 !